atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTask.java

```diff
@@ -69,9 +71,14 @@ public long runOneIteration(SweepableBucket sweepableBucket) {
         long sweepTimestampForIteration =
                 Sweeper.of(sweepableBucket.bucket().shardAndStrategy()).getSweepTimestamp(specialTimestampsSupplier);
         long bucketStartTimestamp = sweepableBucket.timestampRange().startInclusive();
+
+        // TODO(mdaudali): we do need a way to report back to the coordinator about what happened when we tried to
+        //  sweep, so that the autoscaler can adjust as necessary and we can centralise metric reporting.
         if (sweepTimestampForIteration <= bucketStartTimestamp) {
             // This means that the sweep timestamp has not entered this partition yet, so we do not need to process
             // anything. Note that sweep timestamps are exclusive, so <= is correct.
+            targetedSweepMetrics.registerOccurrenceOf(
+                    sweepableBucket.bucket().shardAndStrategy(), SweepOutcome.NOTHING_TO_SWEEP);
```

Comment:
I guess nowadays we already do it for the shard, though one of the things I didn't like when thinking about this one was that it's more _this specific bucket_ having nothing to sweep, and so the reader of this code needs to bear that in mind.